go-controller/pkg/util/policy.go

```diff
@@ -0,0 +1,85 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+// This file contains the util functions commonly used by
+// network policy and admin network policy controllers
+// NOTE: The pieces that are ACL utils must go into libovsdb/utils/acl.go file instead
+// This is only for Policy utils
+
+// ACL L4 Match Construct Utils
+const (
+	// emptyProtocol is used to create ACL for gressPolicy that doesn't have port policies hence no protocols
+	EmptyProtocol = "None"
+)
+
+// GetPortProtocol returns the OVN syntax-specific protocol value for a v1.Protocol K8s type
+func GetPortProtocol(proto v1.Protocol) string {
+	var protocol string
+	switch proto {
+	case v1.ProtocolTCP:
+		protocol = "tcp"
+	case v1.ProtocolSCTP:
+		protocol = "sctp"
+	case v1.ProtocolUDP:
+		protocol = "udp"
+	}
+	return protocol
+}
+
+type PolicyPort struct {
+	Protocol string
+	Port     int32 // will store startPort if its a range
+	EndPort  int32
+}
+
+// for a given ingress/egress rule, captures all the provided port ranges and
+// individual ports
+type gressPolicyPorts struct {
+	portList  []string // list of provided ports as string
+	portRange []string // list of provided port ranges in OVN ACL format
+}
+
+func GetProtocolPortsMap(rulePorts []*PolicyPort) map[string]*gressPolicyPorts {
+	gressProtoPortsMap := make(map[string]*gressPolicyPorts)
+	for _, pp := range rulePorts {
+		gpp, ok := gressProtoPortsMap[pp.Protocol]
+		if !ok {
+			gpp = &gressPolicyPorts{portList: []string{}, portRange: []string{}}
+			gressProtoPortsMap[pp.Protocol] = gpp
+		}
+		if pp.EndPort != 0 && pp.EndPort != pp.Port {
+			gpp.portRange = append(gpp.portRange, fmt.Sprintf("%d<=%s.dst<=%d", pp.Port, pp.Protocol, pp.EndPort))
+		} else if pp.Port != 0 {
+			gpp.portList = append(gpp.portList, fmt.Sprintf("%d", pp.Port))
+		}
+	}
+	return gressProtoPortsMap
+}
+
+func GetL4Match(protocol string, ports *gressPolicyPorts) string {
```

Comment:
I think we can avoid exporting two methods here for now. We could have instead a single exported method
```
func GetL4MatchesFromNetworkPolicyPorts(networkPolicyPorts) []string
```
internally you can handle that however you want.

Reply:
I like this! done!